xtext/org.icyphy.linguafranca/src/org/icyphy/generator/TypeScriptGenerator.xtend

```diff
@@ -248,12 +248,14 @@ class TypeScriptGenerator extends GeneratorBase {
 
             val returnCode = protoc.executeCommand()
             if (returnCode == 0) {
-                val nameSansProto = topLevelName.substring(0, topLevelName.length - 6)
-                targetConfig.compileAdditionalSources.add(
-                this.fileConfig.getSrcGenPath.resolve(nameSansProto + ".pb-c.c").toString)
-
-                targetConfig.compileLibraries.add('-l')
-                targetConfig.compileLibraries.add('protobuf-c')
+                // FIXME: this code makes no sense. It is removing 6 chars from a file with a 3-char extension
```

Comment:
Isn't the filename here a foo.proto file?

Reply:
No, it isn't; I double checked. We have a test `ProtoNoPacking.lf` that passes with this code commented out. `TypeScriptGenerator` needs quite a bit of revision in general, particularly when it comes to the handling of federations.